crates/vendor/morphic_lib/src/analyze.rs

```diff
@@ -256,7 +259,6 @@ impl<'a> ForwardState<'a> {
         // TODO: Optimize this so that it does not traverse the whole parent chain when the heap
         // cell is guaranteed to not have any back ref annotations before a certain point (i.e.,
         // when we have some information about when the heap cell was created).
-        // TODO: Remove query points from back ref sets when they are set to 'DirectTouch'.
```

Comment:
This is the TODO that to my reading means what I did is totally fine.

Reply:
Based on the TODO, I wonder if they could also be removed from other backref sets instead of just the specific one I drained.